SalesforceCommerceCloud/pwa-kit
# @W-16567765 Do Not Track UI

Add the modal for DNT in Template Retail React App

# Description



# Types of Changes




> Breaking changes include:
>
> - Removing a public function or component or prop
> - Adding a required argument to a function
> - Changing the data type of a function parameter or return value
> - Adding a new peer dependency to `package.json`

# Changes

- (change1)

# How to Test-Drive This PR

- (step1) Go to https://runtime-admin-n16.mobify-storefront.com/ and see that the consent form shows up
- (step2) Click outside the modal to see that the modal disappears. Check Developer Tools -> Application -> Cookies to see that dw_dnt is created with its value to 1
- (step3) Delete the cookie and refresh to see that the modal reappears. Click on 'Accept'. Check that dw_dnt with value set to 0 is created in cookies.
- (step4) Delete the cookie and click on another page. See that modal reappears. Click on 'Decline' to see that dw_dnt is created with 1. 
- (step5) Delete the cookie and refresh. Click on the 'X' at the modal to see that dw_dnt with 0 is created. 
- (step6) Go to another page and see that the modal does not pop up again


# Checklists




## General


## Accessibility Compliance

You must check off all items in **one** of the follow two lists:


_or..._


## Localization
